### ISSUE TITLE
Dataset Distillation via Factorization

### ISSUE THREAD

**Issue by reyllama** [reyllama/paper-reviews#207]
# TL; DR

- Maintain inter-sample diversity and discriminability in the synthetic set by factorizing common information and diversity to maximize dataset distillation performance.

# Motivation

- Previous methods treated synthetic samples independently (*i.e., simply optimize them pixel-level*), which limits the overall informativeness of the distilled set.

# Formulation

![image](https://user-images.githubusercontent.com/46308407/215388766-aed68e41-387d-4d89-939e-b0806467830e.png)

- **KEY IDEA**: the base encodes core information (in each class) and hallucinators manipulate their styles to obtain diversity.

![image](https://user-images.githubusercontent.com/46308407/215389028-f8564e7c-7f3b-482b-a008-6586149e2a0a.png)

### Factorization

- As opposed to prior art that had synthetic samples with matching dimensions as the real data, they can keep a smaller size features (bases) and upsample them using CNN hallucinators.
- Hallucinators are style-transfer networks, with encoder-modulation-decoder format. The modulation is learnable AdaIN, where mu and sigma are simply learned network parameters.

### Adversarial Training

- To maximize the information gain from having multiple hallucinators, we want them to re-style bases as orthogonally as possible.
- To assure this behavior, train an additional feature extractor (which is part of the downstream network) and enforce a min-max game.
- MIN: given a same base, the extractor would want to embed two synthetic samples (different hallucinators) as close as possible (because the same base means the same class).
- MAX: The hallucinators want them to be mapped as far as possible in the extractor's feature space.
- On top of simple min-max, of course, the feature extractor should perform well on the target downstream, with an additional output layer that matches the class dimension.

### Objectives

- Synthetic Set (L_DD can be anything but here they used MTT)
![image](https://user-images.githubusercontent.com/46308407/215389881-a13e9412-b4a5-40ef-b303-ce6d52752912.png)
![image](https://user-images.githubusercontent.com/46308407/215389904-d3ae8d89-e848-49ab-af4d-c8cdf5ac0aac.png)
![image](https://user-images.githubusercontent.com/46308407/215389916-66811ad4-86e3-41ba-9e1d-c65fd87e2c09.png)

- Feature Extractor
![image](https://user-images.githubusercontent.com/46308407/215389972-cf51563b-ab00-4182-96b4-70b54fc4277b.png)
![image](https://user-images.githubusercontent.com/46308407/215389986-aac3cba8-1e05-4b93-b955-d1d56d54f726.png)

# Evaluation

![image](https://user-images.githubusercontent.com/46308407/215390062-5efcc179-b912-4359-b02e-329eb0471b24.png)
![image](https://user-images.githubusercontent.com/46308407/215390356-be397ce2-7330-4cb5-b493-cf8e182180d3.png)
![image](https://user-images.githubusercontent.com/46308407/215390446-e9fd15f9-5c2a-4bd6-b5e5-65f8a98c5bf4.png)
![image](https://user-images.githubusercontent.com/46308407/215390543-c8307ed0-e174-4bf6-9fd1-449ca8b94af5.png)

- Experiments are solid, mostly focused on delivering the robustness of this framework under different settings.

# Inspiration

- The high-level motivation to reason about inter-sample correlation/redundancy to acquire maximally informative synthetic set makes much sense.
- The way they factorize samples into common, salient information (for each class like prototype) and different styles (appearances or stochasticity even) is amusing.
- How they posed information-maximizing set construction problem as a min-max problem (similar to GANs) is interesting.
- I think there are much room for improvement based on their core insight to minimize sample redundancy (and the idea of class prototype).
- Still wondering if this whole dataset distillation pipeline can be embedded in the latent space of an off-the-shelf generative model.
